Packs/Fortimail/Integrations/Fortimail/Fortimail.py

```diff
@@ -0,0 +1,2529 @@
+import http
+from functools import wraps
+from typing import Callable, Tuple
+
+import demistomock as demisto  # noqa: F401
+import requests.utils
+import urllib3
+from CommonServerPython import *
+
+urllib3.disable_warnings()
+
+DEFAULT_COOKIE = {"cookie_name": None, "cookie_value": None}
+AUTHORIZATION_ERROR = (
+    "Authorization Error: invalid username or password or too many login attempts."
+    "Please check credentials or wait and try again."
+)
+RP_DIRECTION_MAPPER = {"Inbound": 1, "Outbound": 2}
+BOOLEAN_MAPPER = {"enable": True, "disable": False}
+RP_AUTH_MAPPER = {"radius": 1, "ldap": 2, "pop3": 3, "imap": 4, "samtp": 5}
+RP_PATTERN_TYPE_MAPPER = {
+    "User (wildcard)": 0,
+    "User (regex)": 4,
+    "Email address group": 3,
+    "LDAP group": 2,
+}
+AC_PATTERN_TYPE_MAPPER = {
+    "External": 5,
+    "Internal": 4,
+    "Email Group": 2,
+    "LDAP Group": 3,
+    "LDAP Verification": 6,
+    "Regular Expression": 1,
+    "User Defined": 0,
+}
+AC_AUTH_MAPPER = {
+    "Any": 0,
+    "Not Authenticated": 2,
+    "Authenticated": 1,
+}
+AC_SENDER_IP_TYPE_MAPPER = {
+    "IP/Netmask": 0,
+    "IP Group": 1,
+    "GeoIP Group": 2,
+    "ISDB": 3,
+}
+AC_ACTION_MAPPER = {
+    "Safe & Relay": 1,
+    "Safe": 5,
+    "Relay": 2,
+    "Receive": 6,
+    "Reject": 3,
+    "Discard": 4,
+}
+IP_ACTION_MAPPER = {
+    "Scan": 0,
+    "Fail Temporarily": 2,
+    "Reject": 1,
+    "Proxy bypass": 4,
+}
+DESTINATION_MAPPER = {
+    "IP/Netmask": 1,
+    "IP Group": 2,
+}
+OUTPUT_PREFIX_MAPPER = {
+    "antispam_profile": "AntispamProfile",
+    "geoip_group": "GeoIPgroup",
+    "tls_profile": "TLSprofile",
+    "ip_group": "IPGroup",
+    "ip_group_member": "IPGroup",
+    "email_group": "EmailGroup",
+    "email_group_member": "EmailGroup",
+    "system_safe_block": "SystemList",
+    "ip_policy": "IPPolicy",
+    "access_control": "AccessControl",
+    "recipient_policy": "RecipientPolicy",
+    "ldap_group": "LDAPprofile",
+    "antivirus_profile": "AntivirusProfile",
+    "content_profile": "ContentProfile",
+    "ip_pool": "IPPool",
+    "session_profile": "SessionProfile",
+    "radius_auth_profile": "RadiusAuthProfile",
+    "pop3_auth_profile": "Pop3AuthProfile",
+    "imap_auth_profile": "ImapAuthProfile",
+    "smtp_auth_profile": "SmtpAuthProfile",
+    "resource_profile": "ResourceProfile",
+    "pki_user": "PKIuser",
+}
+MOVE_ACTION = "14"
+REPLACE_ACTION = "10"
+
+
+def validate_authentication(func: Callable) -> Callable:
+    """Decorator to manage authentication for API requests.
+    This decorator first tries to execute the provided function using an existing authentication cookie
```

Comment:
The expiration time for the session is defined by the users and there is no way to know what is the value.
Also, I want to avoid overloading the login endpoint. 
After a few calls in a row, Fortimal blocks me for a while.